flink-table-store-connector/src/main/java/org/apache/flink/table/store/connector/TableStore.java

```diff
@@ -271,43 +264,38 @@ private long discoveryIntervalMills() {
             return options.get(CONTINUOUS_DISCOVERY_INTERVAL).toMillis();
         }
 
-        private FileStoreSource buildFileSource(boolean isContinuous) {
-            FileStore fileStore = buildFileStore();
-
-            boolean latestContinuous = false;
-            if (isContinuous) {
-                LogStartupMode startupMode = logOptions().get(SCAN);
-                latestContinuous = startupMode == LogStartupMode.LATEST;
-            }
+        private FileStoreSource buildFileSource(
+                boolean isContinuous, boolean continuousScanLatest) {
             return new FileStoreSource(
-                    fileStore,
+                    buildFileStore(),
                     primaryKeys.length == 0,
                     isContinuous,
                     discoveryIntervalMills(),
-                    latestContinuous,
+                    continuousScanLatest,
                     projectedFields,
                     partitionPredicate,
                     fieldPredicate);
         }
 
         public Source<RowData, ?, ?> build() {
             if (isContinuous) {
+                LogStartupMode startupMode = logOptions().get(SCAN);
+                boolean latestContinuous = startupMode == LogStartupMode.LATEST;
                 if (logSourceProvider == null) {
-                    return buildFileSource(true);
-                }
-
-                if (isHybrid) {
+                    return buildFileSource(true, latestContinuous);
+                } else {
+                    if (latestContinuous) {
```

Comment:
It seems we need add case for `from_timestamp`.
Only full needs to use hybrid source.